lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -132,17 +135,16 @@ def prepare_data(d, init):
     has_curve = bool(curves)
 
     # Get ScalarMappables.
-    mappabledict = {}
+    mappabledict = []
     for mappable in [*axes.images, *axes.collections]:
         label = mappable.get_label()
         if label == '_nolegend_' or mappable.get_array() is None:
             continue
-        mappabledict[label] = mappable
+        mappabledict.append((label, mappable))
     mappablelabels = sorted(mappabledict, key=cmp_key)
     mappables = []
     cmaps = [(cmap, name) for name, cmap in sorted(cm._cmap_registry.items())]
-    for label in mappablelabels:
-        mappable = mappabledict[label]
+    for label, mappable in mappablelabels:
```

Comment:
```suggestion
    for label, mappable in sorted(mappabledict, key=cmp_key):
```
And delete `mappablelabels = sorted(mappabledict, key=cmp_key)` two lines above.